Java/src/main/java/com/gildedrose/GildedRose.java

```diff
@@ -1,62 +1,32 @@
 package com.gildedrose;
 
+import java.util.List;
+import java.util.stream.Collectors;
+
 class GildedRose {
     Item[] items;
+    List<Updater> updaterList;
 
     public GildedRose(Item[] items) {
         this.items = items;
+        this.updaterList = List.of(new ConstantQualityUpdater(), new IncreasingQualityItemUpdater(), new SimpleDecreasingQualityUpdater(), new BackstageQualityItemUpdater());
     }
 
     public void updateQuality() {
         for (int i = 0; i < items.length; i++) {
-            if (!items[i].name.equals("Aged Brie")
-                    && !items[i].name.equals("Backstage passes to a TAFKAL80ETC concert")) {
-                if (items[i].quality > 0) {
-                    if (!items[i].name.equals("Sulfuras, Hand of Ragnaros")) {
-                        items[i].quality = items[i].quality - 1;
-                    }
-                }
-            } else {
-                if (items[i].quality < 50) {
-                    items[i].quality = items[i].quality + 1;
-
-                    if (items[i].name.equals("Backstage passes to a TAFKAL80ETC concert")) {
-                        if (items[i].sellIn < 11) {
-                            if (items[i].quality < 50) {
-                                items[i].quality = items[i].quality + 1;
-                            }
-                        }
-
-                        if (items[i].sellIn < 6) {
-                            if (items[i].quality < 50) {
-                                items[i].quality = items[i].quality + 1;
-                            }
-                        }
-                    }
-                }
+            Item currentItem = items[i];
+            List<Updater> matchingUpdaterList = updaterList
+            .stream().filter(updater -> updater.getRelevantItemNames().contains(currentItem.name))
```

Comment:
you could return a boolean instead of a list and then you could get rid of the next if condition which is hard to understeand. e.g.
`
@Override
    public boolean isUpdaterFor(Item item) {
        return ...;
    }
`